dask_expr/io/parquet.py

```diff
@@ -693,6 +703,111 @@ def fs(self):
                 fs = type(fs)(**region, **storage_options)
             return fs
 
+    def approx_statistics(self) -> dict:
+        """Return an approximation of a single files statistics.
+
+        This is determined by sampling a few files and averaging their statistics.
+
+        Fields
+        ------
+        num_rows: avg
+        num_row_groups: avg
+        serialized_size: avg
+        columns: list
+            A list of all colum statistics where individual fields are also
+            averaged.
+
+
+        Example
+        -------
+        {
+            'num_rows': 1991129,
+            'num_row_groups': 2.3333333333333335,
+            'serialized_size': 6256.666666666667,
+            'total_byte_size': 118030095,
+            'columns': [
+                {'total_compressed_size': 6284162.333333333,
+                'total_uncompressed_size': 6347380.333333333,
+                'path_in_schema': 'l_orderkey'},
+                {'total_compressed_size': 9423516.333333334,
+                'total_uncompressed_size': 9423063.333333334,
+                'path_in_schema': 'l_partkey'},
+                {'total_compressed_size': 9405796.666666666,
+                'total_uncompressed_size': 9405346.666666666,
+                'path_in_schema': 'l_suppkey'},
+                ...
+            ]
+        }
+
+        Returns
+        -------
+        dict
+        """
+        idxs = self.sample_statistics()
+        files_to_consider = np.array(self._dataset_info["all_files"])[idxs]
+        stats = [_STATS_CACHE[tokenize(finfo)] for finfo in files_to_consider]
+        return _combine_stats(stats)
+
+    def load_statistics(self, files=None, fragments=None, scheduler=None):
+        if files is None:
```

Comment:
Can this potentially load stats twice if called from different places?

Reply:
the statistics are cached in `_STATS_CACHE` by the fileinfo token. This includes the path, last modified, etc.
The `_collect_statistics_plan` is only considering collection of those files which are not already stored in the cache.

This approach would even allow us to store those stats on disk since cache invalidation should be straight forward if not trivial. This would make life easier for folks who are reading the same dataset over and over again.